docs/source/python/getting_involved.rst

```diff
@@ -33,3 +33,42 @@ integrations in other projects, we'd be happy to have you involved:
  * Follow our activity on `JIRA <https://issues.apache.org/jira/browse/ARROW>`_
  * Learn the `Format / Specification
    <https://github.com/apache/arrow/tree/master/format>`_
+
+
+.. _py_arch_overview:
+
+PyArrow Architecture
+--------------------
+
+PyArrow is for the major part a wrapper around the functionalities that
+Arrow C++ implementation provides. The library tries to take what's available
+in C++ and expose it through a user experience that is more pythonic and
+less complex to use. So while in some cases it might be easy to map what's
+in C++ to what's in Python, in many cases the C++ classes and methods are
+used as foundations to build easier to use entities.
+
+.. image:: py_arch_overview.svg
```

Comment:
The image is great and the content corresponding to it is very well structured and understandable!

I would also add alt text to make it more accessible for the readers. Fro example:

```suggestion
.. image:: py_arch_overview.svg
   :alt: Four layers of PyArrow architecture: .py, .pyx, .pxd and low level C++ code.
```